src/Nethermind/Nethermind.Merge.Plugin/Synchronization/UnsafePivotUpdator.cs

```diff
@@ -19,18 +19,22 @@
 
 namespace Nethermind.Merge.Plugin.Synchronization;
 
-public class UnsafePivotUpdator(
-    IBlockTree blockTree,
-    ISyncModeSelector syncModeSelector,
-    ISyncPeerPool syncPeerPool,
-    ISyncConfig syncConfig,
-    IBlockCacheService blockCacheService,
-    IBeaconSyncStrategy beaconSyncStrategy,
-    IDb metadataDb,
-    ILogManager logManager)
-    : PivotUpdator(blockTree, syncModeSelector, syncPeerPool, syncConfig,
-        blockCacheService, beaconSyncStrategy, metadataDb, logManager)
+public class UnsafePivotUpdator : PivotUpdator
 {
+    public UnsafePivotUpdator(
+        IBlockTree blockTree,
+        ISyncModeSelector syncModeSelector,
+        ISyncPeerPool syncPeerPool,
+        ISyncConfig syncConfig,
+        IBlockCacheService blockCacheService,
+        IBeaconSyncStrategy beaconSyncStrategy,
+        IDb metadataDb,
+        ILogManager logManager) : base(blockTree, syncModeSelector, syncPeerPool, syncConfig,
+            blockCacheService, beaconSyncStrategy, metadataDb, logManager)
+    {
+        _beaconSyncStrategy.AllowBeaconHeaderSync();
```

Comment:
`UnsafePivotUpdator` is used by optimism too and it is not expected to allow beacon headers sync from the very beginning there. Would be good to create second, non default constructor with additional flag or just add the flag. At the end - allow beacon heder sync from the beginning for taiko, don't allow for op